src/benchmark/operators/join_benchmark.cpp

```diff
@@ -44,7 +44,8 @@ std::shared_ptr<TableWrapper> generate_table(const size_t number_of_rows) {
                                                EncodingType::Dictionary);
 
   for (ChunkID chunk_id{0}; chunk_id < table->chunk_count(); ++chunk_id) {
-    auto chunk = table->get_chunk(chunk_id);
+    const auto& chunk = table->get_chunk(chunk_id);
```

Comment:
@julianmenzler are you aware you're using one very obscure c++ feature, here? (https://abseil.io/tips/107)

One of @mrks favourites, though.

Reply:
We talked about this in person. This is a bad example for lifetime extension, as we actually do want a copy of the shared_ptr. If `get_chunk` for some reason returned a reference, we could not guarantee that the original shared_ptr does not get destroyed somewhere in between.